apps/system/test/unit/task_manager_test.js

```diff
@@ -881,6 +881,35 @@ suite('system/TaskManager >', function() {
 
   });
 
+  suite('handle multiples home events >', function() {
+    setup(function() {
+      MockStackManager.mStack = [
+        apps['http://sms.gaiamobile.org'],
+        apps['http://game.gaiamobile.org']
+      ];
+      MockStackManager.mCurrent = 0;
+      MockService.currentApp = 'http://sms.gaiamobile.org';
+    });
+
+    teardown(function() {
+      cardsList.innerHTML = '';
+    });
+
+    test('should not add cards-view class when appclosed is received'+
```

Comment:
The unit test is very specific about the implementation. I think we'd do better with a marionette test that asserted the behavior we expect with real events, animation and timing. Can be a follow-up.